### PR TITLE
Add a database connection and execution of the abs compiler

### PR DESCRIPTION
**Pull request by Chabare**
This adds multiple functionalities.

1. Create the database connection
2. Read all necessary tables from the database and convert it to a java model
3. Create ABS-Exports for elements
4. Export all exportable elements to a `Run.abs` file
5. Run the compiler on that file and use it as a data source to display the graph

The database does not completely correspond to the internal model yet, the `Neighbors` table needs to be changed in the database. The new implementation of the `Neighbors` table is used in the internal model.
Some objects are not implemented in the database (e.g. `Strecke`) which are necessary to successfully compile the exported `.abs` file with the ABS-Compiler (absc). Starting the database connection right now results in an "No valid input"-Exception since no Nodes were retrieved from the output of the abs compiler.

### CODE REVIEW

File: src/main/java/com/github/bachelorpraktikum/dbvisualization/config/ConfigKey.java [hunk 1]
```diff
@@ -14,7 +14,12 @@
     initialRestDirectory("rest_initialDirectory"),
     initialRestExecutable("rest_initialExecutable"),
     initialDatabasePort("database_initialPort"),
-    initialDatabaseName("database_initialName");
+    initialDatabaseName("database_initialName"),
+    databaseUsername("database_username"),
+    databasePassword("database_password"),
+    absExportPath("abs_export_path"),
+    absToolchain("abs_toolchain"),
+    experimentalAbsExportForAttributes("experimental_attributes_abs_export");
```
Comment (BjoernPetersen) [inline src/main/java/com/github/bachelorpraktikum/dbvisualization/config/ConfigKey.java]: make sure there are entries in the default.properties file for every key

File: src/main/java/com/github/bachelorpraktikum/dbvisualization/database/Database.java [hunk 2]
```diff
@@ -1,37 +1,359 @@
...
+
+        this.user = user;
+        this.uri = uri;
+        init();
+    }
+
+    /**
+     * Try to connect to the database
+     *
+     * @throws com.zaxxer.hikari.pool.HikariPool.PoolInitializationException if the connection
```
Comment (BjoernPetersen) [inline src/main/java/com/github/bachelorpraktikum/dbvisualization/database/Database.java]: Import the exception and only write the simple name in the JavaDoc

File: src/main/java/com/github/bachelorpraktikum/dbvisualization/database/Database.java [hunk 3]
```diff
@@ -1,37 +1,359 @@
...
+        return edges;
+    }
+
+    /**
+     * Tests whether the database can be accessed over the given URI with the provided {@link
+     * DatabaseUser database user}
+     *
+     * @return Whether a connection can be made
+     */
+
```
Comment (BjoernPetersen) [inline src/main/java/com/github/bachelorpraktikum/dbvisualization/database/Database.java]: Empty line between JavaDoc and function

**Comment by BjoernPetersen** [comment]
Also: the database port from the config seems to be ignored, if some other config entry is missing (which is the case in the default config)

**Comment by Chabare** [comment]
[f32dbf5](https://github.com/Bachelorpraktikum/VisualisierbaR/pull/107/commits/f32dbf5bb5a16db5727dd1ebeb20efccba29d5d3) should fix that the port from the config is ignored when the other values are not set.

File: src/main/java/com/github/bachelorpraktikum/dbvisualization/view/sourcechooser/DatabaseChooserController.java [hunk 4]
```diff
@@ -162,7 +175,34 @@ public ObservableBooleanValue inputChosen() {
     @Nonnull
     @Override
     public DataSource getResource() throws IOException {
-        return null; // TODO this should probably return a SubprocessSource
+        Database database = createDatabase();
+        return new DatabaseSource(database, null);
```
Comment (BjoernPetersen) [inline src/main/java/com/github/bachelorpraktikum/dbvisualization/view/sourcechooser/DatabaseChooserController.java]: createDatabase() can return null (if the chooser is closed before the login window). In that case, this throws a NPE.

**Comment by BjoernPetersen** [comment]
This still prints a full stack trace if the user closes the login window

**Pull request merged**